fedimint-api/src/module/mod.rs

```diff
@@ -198,8 +199,22 @@ where
     }
 }
 
+/// Logic responsible for module's initialization, config generation and validation
+///
+/// Once the module config is ready, the module can be instantiated via `[Self::init]`.
 #[async_trait]
-pub trait FederationModuleConfigGen {
+pub trait ModuleInit {
+    /// Initialize the [`ServerModule`] instance from its config
+    async fn init(
```

Comment:
Possible better name: `from_config`?

Reply:
This method is about actually initializing an instance of a module. `from_config` would sound like a `From::from` constructor for `Self`.

I think it needs to have a "start/init module" in it.